File: XamarinStore.iOS/Views/CustomViewCell.cs
using System;
using MonoTouch.UIKit;

namespace XamarinStore
{

	public class CustomViewCell : UITableViewCell
	{
		UIView child;
		public UIEdgeInsets Padding { get; set; }
		public bool ResizeChild { get; set;}
		public CustomViewCell (UIView child)
			: base (UITableViewCellStyle.Default, "CustomViewCell")
		{
			SelectionStyle = UITableViewCellSelectionStyle.None;
			ResizeChild = true;
			Padding = new UIEdgeInsets ();
			this.child = child;
			var frame = child.Frame;
			frame.Height = frame.Bottom;
			frame.Y = 0;
			Frame = frame;
			this.ContentView.AddSubview (child);
		}

		public override void LayoutSubviews ()
		{
			base.LayoutSubviews ();
			if (!ResizeChild) {
				child.Center = ContentView.Center;
				return;
			}
			var bounds = ContentView.Bounds;
			bounds.X += Padding.Left;
			bounds.Y += Padding.Top;
			bounds.Height -= (Padding.Bottom + Padding.Top);
			bounds.Width -= (Padding.Left + Padding.Right);
			child.Frame = bounds;
		}
	}
}

File: F#/XamarinStore.iOS/Views/CustomViewCell.fs
namespace XamarinStore

open System
open MonoTouch.UIKit

type CustomViewCell(child:UIView) as this =
    inherit UITableViewCell(UITableViewCellStyle.Default, "CustomViewCell")

    do this.SelectionStyle <- UITableViewCellSelectionStyle.None
       let frame = child.Frame
       this.Frame <- new Drawing.RectangleF(frame.X, 0.0f, frame.Width, frame.Bottom)
       this.ContentView.AddSubview child

    member val Padding = new UIEdgeInsets() with get, set
    member val ResizeChild = true with get, set

    override this.LayoutSubviews () =
        base.LayoutSubviews ();
        if not this.ResizeChild then
            child.Center <- this.ContentView.Center
        else
            let bounds = this.ContentView.Bounds;
            child.Frame <- new Drawing.RectangleF(bounds.X + this.Padding.Left,
                                                  bounds.Y + this.Padding.Top,
                                                  bounds.Width - (this.Padding.Left + this.Padding.Right),
                                                  bounds.Height - (this.Padding.Bottom + this.Padding.Top));

